File: Data Structures & Algorithms/valid-palindrome-ii/submission-0.java
class Solution {
    public boolean validPalindrome(String s) {
        int left = 0;
        int right = s.length() - 1;
        int flag = 0;
        while (left < right){
            if (s.charAt(left) != s.charAt(right) && flag==0){
                flag++;
            }
            else if (s.charAt(left) != s.charAt(right) && flag==1){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

File: Data Structures & Algorithms/guess-number-higher-or-lower/submission-0.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        left, right = 0, n
        while True:
            middle = left + ((right - left) // 2)
            res = guess(middle)
            if res > 0:
                left = middle + 1
            elif res < 0:
                right = middle - 1
            else:
                return middle